packages/api/cms-api/src/generator/generate-crud.ts

```diff
@@ -1216,13 +1234,31 @@ function generateResolver({ generatorOptions, metadata }: { generatorOptions: Cr
                 if (${instanceNameSingular}.position < input.position) {
                     await this.${instanceNamePlural}Service.decrementPositions(${
                           positionGroupProps.length
-                              ? `{ ${positionGroupProps.map((prop) => `${prop.name}: ${instanceNameSingular}.${prop.name}`).join(",")} },`
+                              ? `{ ${positionGroupProps
+                                    .map(
+                                        (prop) =>
+                                            `${prop.name}: ${instanceNameSingular}.${prop.name}${
+                                                [ReferenceType.MANY_TO_ONE, ReferenceType.ONE_TO_ONE].includes(prop.reference)
+                                                    ? `${prop.nullable ? `?` : ``}.id`
+                                                    : ``
+                                            }`,
+                                    )
+                                    .join(",")} },`
                               : ``
                       }${instanceNameSingular}.position, input.position);
                 } else if (${instanceNameSingular}.position > input.position) {
                     await this.${instanceNamePlural}Service.incrementPositions(${
                           positionGroupProps.length
-                              ? `{ ${positionGroupProps.map((prop) => `${prop.name}: ${instanceNameSingular}.${prop.name}`).join(",")} },`
+                              ? `{ ${positionGroupProps
+                                    .map(
+                                        (prop) =>
+                                            `${prop.name}: ${instanceNameSingular}.${prop.name}${
+                                                [ReferenceType.MANY_TO_ONE, ReferenceType.ONE_TO_ONE].includes(prop.reference)
+                                                    ? `${prop.nullable ? `?` : ``}.id`
+                                                    : ``
+                                            }`,
+                                    )
+                                    .join(",")} },`
```

Comment:
this is three times the same. probably generate once and reuse it?

Reply:
Does generating the code have any advantage except for not having to repeat yourself three times in the generator? You could also store the code in a variable and use it here.

Reply:
I didn't found a nice place where to put this code... I'll keep it as it is.